src/Ocelot/Configuration/Creator/RoutesCreator.cs

```diff
@@ -76,6 +78,13 @@ public List<Route> Create(FileConfiguration fileConfiguration)
                 .ToList();
         }
 
+        public int CreateTimeout(FileRoute fileRoute, FileGlobalConfiguration globalConfiguration)
+        {
+            return fileRoute.Timeout > 0 
+                ? fileRoute.Timeout.Value 
+                : (globalConfiguration.Timeout ?? DefaultRequestTimeoutSeconds) * 1000;

```

Comment:
```suggestion
                : globalConfiguration.Timeout ?? DefaultRequestTimeoutSeconds;
```
Both values ​​are in seconds!